lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -1005,254 +879,169 @@ public function getCosExpression($value)
      *
      * Computes diff = date1 - date2.
      *
-     * @param string $date1
-     * @param string $date2
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateDiffExpression($date1, $date2)
+    public function getDateDiffExpression(string $date1, string $date2) : string
     {
         throw DBALException::notSupported(__METHOD__);
     }
 
     /**
      * Returns the SQL to add the number of given seconds to a date.
      *
-     * @param string $date
-     * @param int    $seconds
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateAddSecondsExpression($date, $seconds)
+    public function getDateAddSecondsExpression(string $date, string $seconds) : string
```

Comment:
It is a BC break but it should be a `string` by design. When it comes to expressions, every component is an expression as well (e.g. can be a reference to a different field rather than a numeric value). There's no point in restricting it to an `int`, unless some platform-specific corner cases.

Reply:
Addressed in #3498.